src/upper_envelope/fues_jax/fues_jax.py

```diff
@@ -80,35 +94,35 @@ def fast_upper_envelope_wrapper(
             containing refined state- and choice-specific value function.
 
     """
-    min_id = np.argmin(endog_grid)
-    min_wealth_grid = endog_grid[min_id]
-    # These tuning parameters should be set outside. Don't want to touch solve.py now
-    points_to_add = len(endog_grid) // 10
-    num_iter = int(1.2 * value.shape[0])
-    jump_thresh = 2
+    # Set default of n_constrained_points_to_add to 10% of the grid size
+    n_constrained_points_to_add = (
+        endog_grid.shape[0] // 10
+        if n_constrained_points_to_add is None
+        else n_constrained_points_to_add
+    )
+
     # Non-concave region coincides with credit constraint.
     # This happens when there is a non-monotonicity in the endogenous wealth grid
     # that goes below the first point.
     # Solution: Value function to the left of the first point is analytical,
     # so we just need to add some points to the left of the first grid point.
     # We do that independent of whether the condition is fulfilled or not.
     # If the condition is not fulfilled this is points_to_add times the same point.
+    min_id = np.argmin(endog_grid)
+    min_wealth_grid = endog_grid[min_id]
 
     # This is the condition, which we do not use at the moment.
     # closed_form_cond = min_wealth_grid < endog_grid[0]
-    grid_points_to_add = jnp.linspace(min_wealth_grid, endog_grid[0], points_to_add)[
-        :-1
-    ]
+    grid_points_to_add = jnp.linspace(
+        min_wealth_grid, endog_grid[0], n_constrained_points_to_add + 1
+    )[:-1]
     # Compute closed form values
-    values_to_add = vmap(_compute_value, in_axes=(0, None, None, None, None))(
-        grid_points_to_add,
-        expected_value_zero_savings,
-        utility_function,
-        utility_kwargs,
-        disc_factor,
+    values_to_add = vmap(_compute_value, in_axes=(0, None, None, None))(
+        grid_points_to_add, value_function, value_function_args, value_function_kwargs
     )
 
-    # Now determine if we actually had to extend the grid. If not, we just add nans.
+    # Now determine if n_constrained_points_to_addwe actually had to extend the grid.
+    # If not, we just add nans.
```

Comment:
This comment seems a little confusing. Can you rewrite it?

Reply:
Just a typo. Done now,